app/src/main/java/com/github/se/cyrcle/ui/zone/ZoneSelectionScreen.kt

```diff
@@ -1,26 +1,226 @@
 
+import android.content.Context
+import android.util.Log
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.BasicAlertDialog
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Scaffold
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
 import com.github.se.cyrcle.R
+import com.github.se.cyrcle.model.address.AddressViewModel
+import com.github.se.cyrcle.model.map.MapViewModel
+import com.github.se.cyrcle.model.map.MapViewModel.LocationPickerState
+import com.github.se.cyrcle.model.zone.Zone
+import com.github.se.cyrcle.ui.addParking.location.overlay.Crosshair
+import com.github.se.cyrcle.ui.addParking.location.overlay.RectangleSelection
+import com.github.se.cyrcle.ui.map.MapConfig
 import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.theme.atoms.Button
+import com.github.se.cyrcle.ui.theme.atoms.InputText
 import com.github.se.cyrcle.ui.theme.atoms.Text
 import com.github.se.cyrcle.ui.theme.molecules.TopAppBar
+import com.mapbox.geojson.BoundingBox
+import com.mapbox.maps.MapView
+import com.mapbox.maps.extension.compose.DisposableMapEffect
+import com.mapbox.maps.extension.compose.MapboxMap
+import com.mapbox.maps.plugin.gestures.gestures
 
 /** Screen where users can select a new zone to download. */
 @Composable
-fun ZoneSelectionScreen(navigationActions: NavigationActions) {
+fun ZoneSelectionScreen(
+    navigationActions: NavigationActions,
+    mapViewModel: MapViewModel,
+    addressViewModel: AddressViewModel // Possibility to use this to suggest a name for the zone.
+) {
+  val showAlertDialogPickName = remember { mutableStateOf(false) }
+  val boundingBox = remember { mutableStateOf<BoundingBox?>(null) }
+  val zoneName = remember { mutableStateOf("") }
+  val mapView = remember { mutableStateOf<MapView?>(null) }
+  val locationPickerState by
+      mapViewModel.locationPickerState
+          .collectAsState() // state representing where the user is in the location selection
+  // process
+  mapViewModel
```

Comment:
Would it not be better to move this function call into the `LaunchedEffect(Unit)` ? I'm not to familiar with this so I'm not sure if that would be better?

Reply:
It doesn't work, because it launch a new coroutine and the map is created before the camera is updated, making the call useless